src/dune_lang/action.ml

```diff
@@ -259,9 +259,8 @@ let sw = String_with_vars.decode
 let cstrs_dune_file t =
   let open Decoder in
   [ ( "run"
-    , let+ prog = sw
-      and+ args = repeat sw in
-      Run (prog, args) )
+    , let+ args = repeat Slang.decode in
```

Comment:
Why is this repeated?

Reply:
a `Slang.t` is a single term in a command - not the entire command

Reply:
What confused me reading this is that `slang` is supposed to be about encoding lists of strings but here you are only decoding singletons?

Reply:
Slang expressions evaluate to lists of strings but I wouldn't say slang is supposed to be about encoding lists of strings. E.g. there's no syntax for lists of strings in slang. When you write `(run echo hello world)` "echo", "hello", and "world" are 3 different slang expressions.

Do you think I should write a comment to clarify this here?

Reply:
@gridbugs That makes more sense. I think a comment in slang.mli would help clear this up, thanks!